cmd/containerboot/kube.go

```diff
@@ -23,6 +23,54 @@ import (
 	"time"
 )
 
+// checkSecretPermissions checks that the current pod has permission to read,
+// write and patch secretName in the cluster.
+func checkSecretPermissions(ctx context.Context, secretName string) error {
+	for _, verb := range []string{"get", "update", "patch"} {
+		sar := map[string]any{
+			"apiVersion": "authorization.k8s.io/v1",
+			"kind":       "SelfSubjectAccessReview",
+			"spec": map[string]any{
+				"resourceAttributes": map[string]any{
+					"namespace": kubeNamespace,
+					"verb":      verb,
+					"resource":  "secrets",
+					"name":      secretName,
+				},
+			},
+		}
+		bs, err := json.Marshal(sar)
+		if err != nil {
+			return err
+		}
+		req, err := http.NewRequest("POST", "/apis/authorization.k8s.io/v1/selfsubjectaccessreviews", bytes.NewReader(bs))
+		if err != nil {
+			return err
+		}
+		resp, err := doKubeRequest(ctx, req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		bs, err = io.ReadAll(resp.Body)
+		if err != nil {
+			return err
+		}
+		var res struct {
+			Status struct {
+				Allowed bool `json:"allowed"`
+			} `json:"status"`
+		}
+		if err := json.Unmarshal(bs, &res); err != nil {
+			return err
+		}
+		if !res.Status.Allowed {
+			return fmt.Errorf("missing permission: cannot %s secret %q", "patch", secretName)
```

Comment:
I suspect the "patch" constant is supposed to be `verb`

Reply:
Oops, the sedimentary layers of iteration are poking through. Fixed.